nltk/internals.py

```diff
@@ -108,12 +108,13 @@ def java(cmd, classpath=None, stdin=None, stdout=None, stderr=None, blocking=Tru
 
     :raise OSError: If the java command returns a nonzero return code.
     """
-    if stdin == 'pipe':
-        stdin = subprocess.PIPE
-    if stdout == 'pipe':
-        stdout = subprocess.PIPE
-    if stderr == 'pipe':
-        stderr = subprocess.PIPE
+
+    subprocess_output_dict = {'pipe': subprocess.PIPE, 'stdout': subprocess.STDOUT, 'devnull': subprocess.DEVNULL}
+
+    stdin = subprocess_output_dict.get(stdin)
```

Comment:
I would replace this (and the rest accordingly) with
```python
stdin = subprocess_output_dict.get(stdin, stdin)
```
so if an integer or something else is passed, it is used.

Reply:
Agreed.